scio-hdfs/src/main/scala/com/spotify/scio/hdfs/HdfsFileStorage.scala

```diff
@@ -0,0 +1,65 @@
+package com.spotify.scio.hdfs
+
+import java.io.{File, InputStream}
+import java.net.URI
+import java.nio.file.Path
+
+import com.spotify.scio.io.FileStorage
+import org.apache.avro.file.SeekableInput
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, PathFilter}
+import org.apache.hadoop.io.compress.CompressionCodecFactory
+
+
+object HdfsFileStorage {
+  def apply(path: String): FileStorage = new HdfsFileStorage(path)
+}
+
+
+private class HdfsFileStorage(protected val path: String) extends FileStorage {
+
+  private val pathFilter = new PathFilter {
+    override def accept(path: org.apache.hadoop.fs.Path): Boolean =
+      !path.getName.startsWith("_") && !path.getName.startsWith(".")
+  }
+
+  override protected def listFiles: Seq[Path] = {
+    val conf = new Configuration()
+    val fs = FileSystem.get(new URI(path), conf)
+
+    fs
+      .listStatus(new org.apache.hadoop.fs.Path(path), pathFilter)
+      .map(status => new File(path + "/" + status.getPath.getName).toPath).toSeq
```

Comment:
there's a better way than string concat right? `status.getPath` doesn't give you the full path?